backoffice/modules/catalog/components/ProductGeneralInformation.tsx

```diff
@@ -15,54 +19,105 @@ type Props = {
 };
 
 const ProductGeneralInformation = ({ register, errors, setValue }: Props) => {
+  //Get ID
+  const router = useRouter();
+  const { id } = router.query;
+
   const [brands, setBrands] = useState<Brand[]>([]);
 
+  //Get product detail
+  const [product, setProduct] = useState<Product>();
+  const [isLoading, setLoading] = useState(false);
+
   useEffect(() => {
     getBrands().then((data) => {
       setBrands(data);
     });
-  }, []);
+  },[])
+
+  useEffect(() => {    
+    // In case of updating we load product base on id
+    if (id) {
+      setLoading(true);
+      getProduct(+id).then((data) => {
+        if (data) {
+          setProduct(data);
+          setLoading(false);
+        } else {
+          //Show error
+          toast(data.detail);
+          location.replace('/catalog/products');
+        }
+      });
```

Comment:
In my opinion, check data is exist or not in .then() of Promise doesn't make sense based on the following reasons below:
1. .then() only fires when HttpStatus of response of API call is 2xx (Status of Promise is Fulfilled)
2. An object (here is data) in JS is truthy, so if statement is always true

My solution for checking the API call failed is to use .catch() because when the response returns HttpStatus code different from 2xx it will return a Promise with status Rejected and .catch() will fire.

Code suggestion:
`
getProduct(+id)
     .then((data) => {
          setProduct(data);
          setLoading(false);
     })
     .catch((error) => {
          toast('check error to get message');
          location.replace('/catalog/products');
     })
`

Reply:
Hi @khoahd7621 thanks for your reviewing